### DISCUSSION TITLE
Short url click stats by datetime

### DISCUSSION THREAD

**Discussion by breesemarkides** [YOURLS/YOURLS#3237] [Q&A]
I'd like to create a dataframe that has short url click stats broken down by datetime. I.e., each click would be an observation with a timestamp.

I could get these data with the url-stats-period plugin and a loop:

```
for i in range (<date link was created in Epoch seconds>, <current datetime in Epoch seconds>):
    until = i+1
    r = requests.get(f'{api_url}?signature={signature}&format=json&action=url-stats-period&shorturl={short_url}&since={i}&until={until}')

##extract clicks and append to dataframe##
```

But that is obviously very wasteful and time-consuming. Am I missing something in the API that will let me see the datetime each click occurred?

**Comment by dgw** [comment]
I think you're talking about something that would be best implemented as your own custom plugin that adds an API action, just like the `url-stats-period` plugin you mentioned already does.

Your own action that takes a shorturl keyword and returns all of the clicks seems to be the best approach.
